### ISSUE TITLE
Syslog returner does not work with Python 2.6

### ISSUE THREAD

**Issue by jbadson** [saltstack/salt#40688]
### Description of Issue/Question
The syslog returner raises a TypeException when calling syslog.openlog in Python 2.6.  This is because the syslog.openlog function does not accept keyword arguments in that Python version.

While I'd like to dismiss this because Python 2.6 is deprecated, there are unfortunately a lot of enterprises still stuck on CentOS 6, and Salt doesn't yet provide a way to specify which Python interpreter to use.

### Setup
Fresh salt installation from using Yum from the saltstack repo.

### Steps to Reproduce Issue
This affects any command that uses the syslog returner. The command executes and returns to stdout normally, but nothing ends up in syslog. 

Example:
```
salt-call --return syslog status.loadavg
local:
    ----------
    1-min:
        0.0
    15-min:
        0.06
    5-min:
        0.06
```

From /var/log/salt/minion:
```
017-04-13 20:43:03,649 [salt.utils.schedule                                  ][ERROR   ][1726] Unhandled exception running status.diskusage
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 855, in handle_func
    self.returners[ret_str](ret)
  File "/usr/lib/python2.6/site-packages/salt/returners/syslog_return.py", line 198, in returner
    syslog.openlog(logoption=logoption)
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
```

**Comment by gtmanfred** [comment]
Thanks for the PR for this!
Daniel

**Comment by cachedout** [comment]
Closed via #40689
